File: snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/entity/ClientSessionEntity.java
package com.snowplowanalytics.snowplow.entity;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.snowplowanalytics.snowplow.internal.constants.Parameters;
import com.snowplowanalytics.snowplow.internal.constants.TrackerConstants;
import com.snowplowanalytics.snowplow.payload.SelfDescribingJson;

import java.util.Map;

public class ClientSessionEntity extends SelfDescribingJson  {

    private final Map<String, Object> values;

    public ClientSessionEntity(@NonNull Map<String, Object> values) {
        super(TrackerConstants.SESSION_SCHEMA, values);

        this.values = values;
    }

    @Nullable
    public String getUserId() {
        return (String) values.get(Parameters.SESSION_USER_ID);
    }
}

File: snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/entity/ClientSessionEntity.kt
/*
 * Copyright (c) 2015-2023 Snowplow Analytics Ltd. All rights reserved.
 *
 * This program is licensed to you under the Apache License Version 2.0,
 * and you may not use this file except in compliance with the Apache License Version 2.0.
 * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the Apache License Version 2.0 is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
 */

package com.snowplowanalytics.snowplow.entity

import com.snowplowanalytics.core.constants.Parameters
import com.snowplowanalytics.core.constants.TrackerConstants
import com.snowplowanalytics.snowplow.payload.SelfDescribingJson

/**
 * Used to represent session information
 */
class ClientSessionEntity(private val values: Map<String, Any?>) :
    SelfDescribingJson(TrackerConstants.SESSION_SCHEMA, values) {

    val userId: String?
        get() = values[Parameters.SESSION_USER_ID] as String?
}
